### DISCUSSION TITLE
Eclipse syntax errors with alignas usage (but compiles fine)

### DISCUSSION THREAD

**Discussion by randyK-HH** [dvidelabs/flatcc#215] [Q&A]
I am using Flatbuffers in an Eclipse based embedded application and functionally it works perfectly. 

One hiccup though I have come across is that Eclipse does not seem to like the "alignas(X)" key words in front of the struct field entries.

As an example, take the following struct which is lifted from the specialized reader file generated by the flatCC compiler.

```
struct MyDataNS_DataStruct {
    alignas(4) uint8_t value1;
    alignas(1) uint8_t value2;
    alignas(2) uint16_t value3;
    alignas(4) uint32_t value4;
};
```

Each line is declared a "Syntax Error" by Eclipse. Then, future references to the entries result in a "Field could not be resolved" error. However, as mentioned, it still builds/compiles just fine. (If I remove the alignas(X) the syntax error goes away.)

Is there any way to eliminate these error messages from showing up?

Note: I have PORTABLE_C11_ALIGNED_ALLOC defined to zero in my symbol table.

**Comment by mikkelfj** [comment]
I have not tried this since the dawn of flatcc because it has not been necessary until now, but you can replace alignas with pragma pack(1) which also inserts explicit zero padding fields in structs. I'm not sure if flatcc still compiles with it, and if it does, I'm not sure the output is correct, but you could try to use it while editing and then generate the final output as usual.

Unfortunately this particular setting is not guarded by #ifndef, so you have to edit the config file:
https://github.com/dvidelabs/flatcc/blob/a889e487609ec0f2b6aa9b6a920bc8129a8b547f/config/config.h#L390

**Comment by mikkelfj** [comment]
Another option might be to edit the portable file the provides the alignas feature portably, and if eclipse is defined somehow, define alignas as the empty macro. Depending on how the syntax checker works, this might resolve the issue.
